src/components/InterruptedVideosGallery.tsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react'
+import { RouteComponentProps } from '@reach/router'
+
+import { useQuery } from '@apollo/client'
+import { usePersonalData } from '@/hooks'
+import { GET_VIDEOS_WITH_IDS } from '@/api/queries'
+
+import { GetVideosWithIds, GetVideosWithIdsVariables } from '@/api/queries/__generated__/GetVideosWithIds'
+
+import { ErrorFallback, VideoGallery } from '@/components'
+
+interface Lookup {
+  [key: string]: number
+}
+
+const InterruptedVideosGallery: React.FC<RouteComponentProps> = () => {
+  const {
+    state: { watchedVideos },
+  } = usePersonalData()
+  const interruptedVideosState = watchedVideos.filter((video) => video.__typename === 'INTERRUPTED' && video.timestamp)
+  const interruptedVideosId = interruptedVideosState.map((video) => video.id)
+
+  const { loading, data, error, refetch } = useQuery<GetVideosWithIds, GetVideosWithIdsVariables>(GET_VIDEOS_WITH_IDS, {
+    variables: { ids: interruptedVideosId },
+  })
+
+  const videoTimestampsMap: Record<string, number> = interruptedVideosState.reduce((acc: Lookup, video) => {
+    if (video.__typename === 'INTERRUPTED') {
+      acc[video.id] = video.timestamp || 0
+      return acc
+    }
+    return acc
+  }, {})
```

Comment:
Do we need that `acc: Lookup`? Is `acc` type not figured out correctly? If not, we can instead do something like this:

```suggestion
  const videoTimestampsMap = interruptedVideosState.reduce((acc, video) => {
    if (video.__typename === 'INTERRUPTED') {
      acc[video.id] = video.timestamp || 0
      return acc
    }
    return acc
  }, {} as Record<string, number>)
```